Project Management Automation: Commit-to-master fails addMilestone task

Related: #20058

**Describe the bug**

Direct commits to the `master` branch will result in a failing action for the `addMilestone` task.

See: https://github.com/WordPress/gutenberg/runs/436475328

```
main: Received event = 'push', action = 'undefined'
main: Starting task addMilestone
##[error]main: Task addMilestone failed with error: TypeError: object null is not iterable (cannot read property Symbol(Symbol.iterator))
main: All done!
##[error]Node run failed with exit code 1
```

While commits to master are discouraged and should be rare occurrences, the action should be able to handle this gracefully.

Based on the error message, I suspect that the issue is that `String#match` will return `null` when there is no match:

https://github.com/WordPress/gutenberg/blob/38f4914149bc6fecbeb45e86ed29b2c101ca65fc/packages/project-management-automation/lib/add-milestone.js#L44

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#Return_value

Before deconstructing the array members, this should check that the result would be non-`null`.

--- a/packages/project-management-automation/lib/add-milestone.js
+++ b/packages/project-management-automation/lib/add-milestone.js
@@ -41,7 +41,8 @@ async function addMilestone( payload, octokit ) {
 		return;
 	}
 
-	const [ , prNumber ] = payload.commits[ 0 ].message.match( /\(#(\d+)\)$/m );
+	const match = payload.commits[ 0 ].message.match( /\(#(\d+)\)$/m );
+	const prNumber = match && match[ 1 ];
 	if ( ! prNumber ) {
 		debug( 'add-milestone: Commit is not a squashed PR. Aborting' );
 		return;

